flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/table/MongoDynamicTableFactoryTest.java

```diff
@@ -0,0 +1,340 @@
+        Map<String, String> properties = getRequiredOptions();
+        properties.put(BUFFER_FLUSH_MAX_ROWS.key(), "1001");
+        properties.put(BUFFER_FLUSH_INTERVAL.key(), "2min");
+        properties.put(DELIVERY_GUARANTEE.key(), "at-least-once");
+        properties.put(SINK_MAX_RETRIES.key(), "5");
+        properties.put(SINK_RETRY_INTERVAL.key(), "2s");
+
+        DynamicTableSink actual = createTableSink(SCHEMA, properties);
+
+        MongoConnectionOptions connectionOptions =
+                MongoConnectionOptions.builder()
+                        .setUri("mongodb://127.0.0.1:27017")
+                        .setDatabase("test_db")
+                        .setCollection("test_coll")
+                        .build();
+        MongoWriteOptions writeOptions =
+                MongoWriteOptions.builder()
+                        .setBatchSize(1001)
+                        .setBatchIntervalMs(TimeUnit.MINUTES.toMillis(2))
+                        .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
+                        .setMaxRetries(5)
+                        .setRetryIntervalMs(TimeUnit.SECONDS.toMillis(2))
+                        .build();
+
+        MongoDynamicTableSink expected =
+                new MongoDynamicTableSink(
+                        connectionOptions,
+                        writeOptions,
+                        null,
+                        SCHEMA.getPrimaryKey().isPresent(),
+                        SCHEMA.toPhysicalRowDataType(),
+                        MongoKeyExtractor.createKeyExtractor(SCHEMA));
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void testMongoSinkWithParallelism() {
+        Map<String, String> properties = getRequiredOptions();
+        properties.put("sink.parallelism", "2");
+
+        DynamicTableSink actual = createTableSink(SCHEMA, properties);
+
+        MongoConnectionOptions connectionOptions =
+                MongoConnectionOptions.builder()
+                        .setUri("mongodb://127.0.0.1:27017")
+                        .setDatabase("test_db")
+                        .setCollection("test_coll")
+                        .build();
+
+        MongoWriteOptions writeOptions = MongoWriteOptions.builder().build();
+
+        MongoDynamicTableSink expected =
+                new MongoDynamicTableSink(
+                        connectionOptions,
+                        writeOptions,
+                        2,
+                        SCHEMA.getPrimaryKey().isPresent(),
+                        SCHEMA.toPhysicalRowDataType(),
+                        MongoKeyExtractor.createKeyExtractor(SCHEMA));
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void testMongoValidation() {
+        // fetch size lower than 1
+        Map<String, String> properties = getRequiredOptions();
+        properties.put(SCAN_FETCH_SIZE.key(), "0");
+
+        Map<String, String> finalProperties1 = properties;
```

Comment:
You could make this test a lot nicer with a utility method like `getRequiredOptionsWithSetting(String, String)`.
```
assertThatThrownBy(() -> createTableSource(SCHEMA, getRequiredOptionsWithSetting(SCAN_FETCH_SIZE.key(), "0"))
        .hasStackTraceContaining("The fetch size must be larger than 0.");
```

Bonus points for adding another `assertValidationRejects(String, String,String)` method.

```
assertValidationRejects(SCAN_FETCH_SIZE.key(), "0", "The fetch size must be larger than 0.");
```